### PR TITLE
Issue 28251 replacing api calls 

### LINKED ISSUE #28251: Replace all api calls that retrieve LanguageKey by the a new APIs that return LanguageVariable (Content)
### Parent Issue

https://github.com/dotCMS/core/issues/25603

### User Story

Write a new API Method to retrieve all Language Variables as lightly as possible. No contentlet additional info should be making it into the returned object. Only identifier, key, value, and language.
This way the resulting method can be cached with a smaller memory footprint.  Once this method is in place and caching is properly handled we need to identify the actual use of the old APIs and replace them with the respective calls to the new ones. So we only rely on the language variables stored as content. 
This method should ignore permissions and should content state (working or live) any LanguageVariable should be returned.

### Acceptance Criteria

After completing this task No use of the old LanguageKey API should remain active in code.  
The Old APIS should be marked as deprecated. So they can be easily identified shortly for removal. 
The new APIs method will make it possible to load and cache all Language Variables despite of state and permissions  

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
### Proposed Changes
* A Feature Flag in the form of a method called isLocalizationEnhancementsEnabled has been added to activate/deactivate the functionality.
* The Feature flag is for now on. If we want to release we need to set it to off.
* Added the capability of accessing the language variable in the cache using only the key and the lang ID
* Fixed a bug in the queries introduced when changed to show both live and working content. The bug was making all archived pieces of content appear in the results.
* Fixes on Integration Tests
* Fixes on postman test. That was caused by a dupe key introduced during the postman run. The unique key constraint can be lifted, so we must be prepared for such a situation. 
* New API Calls added next to the old one. The new ones are executed instead of the old APIs only if the feature flag is on.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPI.java [hunk 1]
```diff
@@ -123,13 +137,15 @@ public interface LanguageAPI {
 	 * @param lang
 	 * @return
 	 */
+	@Deprecated(since = "14.05", forRemoval = true)
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPI.java:140]: shouldn't it be 24.05?

Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPI.java:140]: its a date

File: dotcms-integration/src/test/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPITest.java [hunk 2]
```diff
@@ -181,48 +183,91 @@ public void getStringKey_returnKey(){
...
+
+		assertEquals(LanguageUtil.get(new Locale("en", "us"), SYSTEM_PROPERTYFILE_KEY),
+				translatedMap.get(SYSTEM_PROPERTYFILE_KEY));
+	} finally {
+	   System.clearProperty(LOCALIZATION_ENHANCEMENTS_ENABLED);
+	}
+  }
+
+  @Test
+  public void getStringsAsEnhancementsEnabled() throws Exception {
```
Comment (nollymar) [inline dotcms-integration/src/test/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPITest.java:238]: Please add javadoc

File: dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPI.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPI.java]: yes, I changed it

File: dotcms-integration/src/test/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPITest.java (hunk 2 above)
Comment (fabrizzio-dotCMS) [reply dotcms-integration/src/test/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPITest.java:244]: done

**Review by nollymar — APPROVED** [review]

**Review by dsolistorres — APPROVED** [review]

**Pull request merged**